docker/docker_client.go

```diff
@@ -103,11 +103,47 @@ func newDockerClient(ctx *types.SystemContext, ref dockerReference, write bool)
 		tlsc := &tls.Config{}
 
 		if ctx.DockerCertPath != "" {
-			cert, err := tls.LoadX509KeyPair(filepath.Join(ctx.DockerCertPath, "cert.pem"), filepath.Join(ctx.DockerCertPath, "key.pem"))
```

Comment:
I highly doubt anyone was using `--cert-path` since right now it was only possible to send client certificates against a given registry - this change *fixes* this behavior by sending the CA and ,if any, clients certificates (which is what docker does as well).

I believe this was a bug.